src/MongoDB.Driver.Core/Core/NativeLibraryLoader/RelativeLibraryLocatorBase.cs

```diff
@@ -35,31 +36,39 @@ public virtual Assembly GetLibraryBaseAssembly()
 
         public virtual string GetLibraryBasePath()
         {
-            var assembly = GetLibraryBaseAssembly();
-            var codeBase = assembly.CodeBase;
-            var uri = new Uri(codeBase);
-            var absolutePath = uri.AbsolutePath;
-            return Path.GetDirectoryName(absolutePath);
+            // In the nuget package, get the shared library from a relative path of this assembly
+            // Also, when running locally, get the shared library from a relative path of this assembly
+            var assembly = typeof(LibraryLoader).GetTypeInfo().Assembly;
+            var location = assembly.Location;
+            string basePath = Path.GetDirectoryName(location);
+            return Path.GetDirectoryName(basePath);
         }
 
         public abstract string GetLibraryRelativePath(SupportedPlatform currentPlatform);
 
         // private methods
-        private string FindLibraryOrThrow(string basePath, string relativePath)
+        private string FindLibraryOrThrow(string[] basePaths, string[] suffixPaths, string library)
         {
-            var absolutePath = Path.Combine(basePath, relativePath);
-            if (!File.Exists(absolutePath))
+            var failedPaths = new List<string>();
+            foreach (var basePath in basePaths)
             {
-                throw new FileNotFoundException($"Could not find library {absolutePath}.");
+                foreach (var suffix in suffixPaths)
+                {
+                    string path = Path.Combine(basePath, suffix, library);
+                    if (File.Exists(path))
+                    {
+                       return path;
+                    }
+                    failedPaths.Add(path);
+                }
             }
-
-            return absolutePath;
-        }
+            throw new FileNotFoundException("Could not find: " + library + " --\n Tried: " + string.Join(",", failedPaths));
```

Comment:
What is the purpose of `--` in the message?

Maybe use `", "` (with a space after the comma) instead of `","` in `string.Join`.